### DISCUSSION TITLE
Some errors that are not already reported

### DISCUSSION THREAD

**Discussion by markfirmware** [alpinejs/alpine#975] [1. Help] [answered]
[codepen that detects more alpine errors](https://codepen.io/markfirmware/pen/WNGzEMg)

Can this error detection be added to alpine?
Thanks, Mark

**Comment by HugoDF** [comment]
- `x-for/x-if` must be on a template, pretty sure this is already logs a warning
- `x-init requires x-data on same element` this might be reasonable to add, if we only check that DOM elements inside of an Alpine component don't have `x-init` (unless they're the root element), you can try submitting a PR for @calebporzio to have a look.
- `x-for/x-if must be in scope of an element (even this one itself) that has x-data`, this doesn't make sense to me, this warning would only be triggered when you've got an Alpine directive in non-Alpine code, which feels weird to be logging a warning for (not to mention the fact you're scanning unrelated DOM for Alpine directives)

**Reply by markfirmware** [reply]
Is an alpine-only web page supported? It is a fundamental use case, in addition to the use case of just sprinkling x-data. Is adding x-data to body the answer here?

**Reply by KevinBatdorf** [reply]
Might be useful as a VS code plugin or just a standalone plugin to use during development (or in DevTools). 

Something that just parses the code looking for common issues and reports them to the user.

**Reply by markfirmware** [reply]
Would such a plugin work in codepen? As it stands, pasting one div after the body, and adding x-data to the body, implements this reporter readily in codepen.

This begs the next question after what browser do you use: what ide do you use?

**Reply by KevinBatdorf** [reply]
You can write your own JS code to check for common errors:

For example, you could build out this pen to do a bunch of checks, whatever you think might be helpful. You can also put it on github and publish it to NPM for others to use.
https://codepen.io/KevinBatdorf/pen/JjRvdoR

Then to use it, you add that JS file to other html you have (or other pens, or a codepen template)
https://codepen.io/KevinBatdorf/pen/JjRvdoR.js

You can see it in use here:
https://codepen.io/KevinBatdorf/pen/KKgRppY?editors=1111

I did this in 5 minutes so it might be buggy. I just want to demonstrate it. If it's buggy, fork it and fix it :)

**Reply by markfirmware** [reply]
Right. A js-only codepen. Just right. Thanks! I think I will pursue this.

**Comment by markfirmware** [comment, marked as answer]
A js-only codepen to be added when more diagnostics are desired. Thanks @KevinBatdorf!

Thanks @HugoDF - I will continue with your comments in another discussion.

**Reply by HugoDF** [reply]
FYI @markfirmware I've created an issue on the devtools repo to add some warnings for common issues in markup https://github.com/alpine-collective/alpinejs-devtools/issues/141
